ports/libsimpleservo/capi/src/lib.rs

```diff
@@ -18,114 +27,155 @@ use std::os::raw::{c_char, c_void};
 use std::panic::{self, AssertUnwindSafe, UnwindSafe};
 use std::thread;
 
-// #[cfg(target_os = "windows")]
-use libc;
-use winapi::shared;
-use winapi::um::debugapi;
-use winapi::um::handleapi;
-use winapi::um::minwinbase;
-use winapi::um::namedpipeapi;
-use winapi::um::processenv;
-use winapi::um::winbase;
-use winapi::um::winnt;
-
 /// Catch any panic function used by extern "C" functions.
 fn catch_any_panic<F: FnOnce() + UnwindSafe>(function: F) -> bool {
     panic::catch_unwind(function).is_ok()
 }
 
+#[cfg(not(target_os = "windows"))]
+fn redirect_stdout_stderr() -> Result<(), String> {
+    Ok(())
+}
+
+#[cfg(target_os = "windows")]
+fn do_redirect_stdout_stderr() -> Result<(), ()> {
+    // actual implementation that returns Err(()) for problems
```

Comment:
My point was that all of the actual implementation that is currently in redirect_stdout_stderr should go in do_redirect_stdout_stderr instead. This will allow all of the cases that used to be `return false` to be `return Err(())` instead, and then we need one place in redirect_stdout_stderr that does the map_err thing and calls GetLastError.

Reply:
Suggested changes implemented.